packages/agreement-process/src/services/agreementService.ts

```diff
@@ -605,7 +600,7 @@ export async function upgradeAgreementLogic({
     return {
       streamId: upgraded.id,
       events: [
```

Comment:
@beetlecrunch is it possible we missed the `Upgraded` business event?
I would expect a specific event in this case (but it would impact 2 different agreements).
Having a separate `Archived` event may lead to unwanted notifications/logics from consumers

Reply:
Previously, we archived the agreement and created a new one. As a result, I replicated the same behavior using the new events.
I agree that having a separated `Archived` event may lead to unwanted notifications